### ISSUE TITLE
Can't read/access file using the FTP adapter

### ISSUE THREAD

**Issue by DamienPirsy** [thephpleague/flysystem#308]
I can succesfully list a directory content, read/create new directories and even create new files, but I really can't figure out why I'm not able to read existing files (I'm going to implement a download feature). I'm using the Laravel Flysystem adapter, by the way.

This command:
`$dirs = \Flysystem::listContents('Z19633740');`

succesfully returns an array of what's inside /Z19633740/:

```
array(1) {
  [0]=>
  array(9) {
    ["dirname"]=>
    string(9) "Z19633740"
    ["basename"]=>
    string(10) "165138.txt"
    ["extension"]=>
    string(3) "txt"
    ["filename"]=>
    string(6) "165138"
    ["path"]=>
    string(20) "Z19633740/165138.txt"
    ["type"]=>
    string(4) "file"
    ["visibility"]=>
    string(6) "public"
    ["size"]=>
    int(14)
    ["timestamp"]=>
    int(1415104740)
  }
```

Now, I'm tryng to read the content of the file, but all of these fails:

```
\Flysystem::read('Z19633740/165138.txt');
\Flysystem::read('./Z19633740/165138.txt');
\Flysystem::read('./165138.txt');
```

The error returned is (JSON encoded):

`{"error":{"type":"League\\Flysystem\\FileNotFoundException","message":"File not found at path: 165138.txt","file":"\/var\/www\/wfm_smart\/vendor\/league\/flysystem\/src\/Filesystem.php","line":733}}`

What I don't understand is why ` \Flysystem::write('Z19633740/165138.txt', $content)` works but the reverse doesn't. Also, the delete function (and any file-relevant function like getMimetype() )doesn't work, same "File not found at path" error.

Further infos, the application is being developed on an Ubuntu system, but the FTP is on a intranet NAS disk under Windows (can't tell the OS version now) and the directory structure is:

```
/
   Z19633740/
               165138.txt
```

**Comment by denndk** [comment]
Can confirmed this issue using Windows 2008 server with Filezilla FTP server. The STAT command is not implemented in Filezilla. 

The command is sent in the FtpAdapter:
```php
public function getMetadata($path)
    {
       
        if (empty($path) ||  ! ($object = ftp_raw($this->getConnection(), 'STAT ' . $path)) || count($object) < 3) {
            
            return false;
        }

        return $this->normalizeObject($object[1], '');
    }
```

As far as I can tell the STAT command is not the most common implementation. A better solution would be the command LIST.

**Comment by frankdejonge** [comment]
This is fixed in dev-master (wil be released in 1.0.2).
